cli/src/deno.rs

```diff
@@ -1,55 +1,71 @@
-pub struct DenoRuntime {
-    runtime: JsRuntime,
-}
-
-impl Default for DenoRuntime {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
-impl DenoRuntime {
-    /// Create a new Deno runtime.
-    pub fn new() -> Self {
-        // TODO: Add Phylum API methods here.
-        let phylum_api = Extension::builder().build();
-
-        let runtime = JsRuntime::new(RuntimeOptions {
-            module_loader: Some(Rc::new(TypescriptModuleLoader)),
-            extensions: vec![phylum_api],
-            ..Default::default()
-        });
-
-        Self { runtime }
-    }
-
-    /// Execute a JavaScript module from its main entry point.
-    pub async fn run(&mut self, entrypoint: &str) -> Result<()> {
-        let module_specifier = deno_core::resolve_path(entrypoint)?;
-        let module = self
-            .runtime
-            .load_main_module(&module_specifier, None)
-            .await?;
-        let _ = self.runtime.mod_evaluate(module);
-
-        self.runtime.run_event_loop(false).await?;
-
-        Ok(())
-    }
+use deno_runtime::permissions::Permissions;
+use deno_runtime::worker::{MainWorker, WorkerOptions};
+use deno_runtime::{colors, BootstrapOptions};
+
+/// Execute Phylum extension.
+pub async fn run(entry_point: &str) -> Result<()> {
+    let main_module = deno_core::resolve_path(entry_point)?;
+
+    let cpu_count = thread::available_parallelism()
+        .map(|p| p.get())
+        .unwrap_or(1);
+    let bootstrap = BootstrapOptions {
+        cpu_count,
+        runtime_version: env!("CARGO_PKG_VERSION").into(),
+        user_agent: "phylum-cli/extension".into(),
+        no_color: !colors::use_color(),
+        is_tty: colors::is_tty(),
+        enable_testing_features: Default::default(),
+        debug_flag: Default::default(),
+        ts_version: Default::default(),
+        location: Default::default(),
+        unstable: Default::default(),
+        args: Default::default(),
+    };
+
+    let options = WorkerOptions {
+        bootstrap,
+        web_worker_preload_module_cb: Arc::new(|_| unimplemented!("web workers are not supported")),
+        create_web_worker_cb: Arc::new(|_| unimplemented!("web workers are not supported")),
+        module_loader: Rc::new(TypescriptModuleLoader),
+        unsafely_ignore_certificate_errors: Default::default(),
+        should_break_on_first_statement: Default::default(),
+        compiled_wasm_module_store: Default::default(),
+        shared_array_buffer_store: Default::default(),
+        maybe_inspector_server: Default::default(),
+        format_js_error_fn: Default::default(),
+        get_error_class_fn: Default::default(),
+        origin_storage_dir: Default::default(),
+        broadcast_channel: Default::default(),
+        source_map_getter: Default::default(),
+        root_cert_store: Default::default(),
+        blob_store: Default::default(),
+        extensions: Default::default(),
+        stdio: Default::default(),
+        seed: Default::default(),
```

Comment:
I think the use `Default::default()` makes this code harder to read... Here are a few examples to explain:

* `should_break_on_first_statement` is a `bool`, so it gets the default value of `false`. But `should_break_on_first_statment: false` would be easier to read.
* `seed` is `Option<u64>`, so it gets a value of `None` which makes sense because we don't care about the seed. But if `seed` were just `u64`, then that would be a seed of `0` which would be bad. Writing `seed: None` makes it clear without having to look at the type.

If `WorkerOptions` implemented `Default`, then we could use `..Default::default()` and trust that the developers thought about the default for each field when implementing that trait. Without that, we have to do the thinking ourselves. And if we have to think about the actual values, we might as well write out the values (i.e., mostly `None`).

Reply:
I'd argue using `Default::default()` here makes things **significantly** easier to understand. The goal is to make my intentions here clear and obvious, none of these fields are set to any specific value. I just wanted to set these values to **any** value that wouldn't hinder us. I didn't set `should_break_on_first_statement` to `false` with this, I just tried not setting it to anything else and see if it does the trick for us.

> seed is Option<u64>, so it gets a value of None which makes sense because we don't care about the seed. But if seed were just u64, then that would be a seed of 0 which would be bad. Writing seed: None makes it clear without having to look at the type.

I mean a seed of 0 is still a valid seed and it would be just as valid as any other fixed seed we might set this to. Again, the decision was to **NOT** worry about the seed and have it set to just some random value and move on. I think default here exactly proves the point that I did not worry about setting any seed, while setting the seed to `None` might look like I intentionally set it to this value.

> Without that, we have to do the thinking ourselves. And if we have to think about the actual values, we might as well write out the values (i.e., mostly None).

The writer has to do the thinking, but the reader doesn't. And if we add explicit options for everything the reader has to do the thinking.

If there are specific options you want to assign certain values, like `seed` for example, we can do that. But I don't think it provides a value to set fields we do not care about to a specific value, since that misleads the reader more than it helps.